--- a/src/app/services/services.module.ts
+++ b/src/app/services/services.module.ts
@@ -120,20 +120,16 @@ const notificationsService = new NotificationsService(userService, syncService,
 const environmentService = new EnvironmentService(apiService, storageService, notificationsService);
 const auditService = new AuditService(cryptoFunctionService, apiService);
 
-const analytics = new Analytics(window, () => platformUtilsService.isDev() || platformUtilsService.isSelfHost(),
+const analytics = new Analytics(window, () => platformUtilsService.isDev() || platformUtilsService.isSelfHost() || true,
     platformUtilsService, storageService, appIdService);
 containerService.attachToWindow(window);
 
 export function initFactory(): Function {
     return async () => {
         await (storageService as HtmlStorageService).init();
-        const isDev = platformUtilsService.isDev();
-        if (!isDev && platformUtilsService.isSelfHost()) {
-            environmentService.baseUrl = window.location.origin;
-        } else {
-            environmentService.notificationsUrl = isDev ? 'http://localhost:61840' :
-                'https://notifications.bitwarden.com'; // window.location.origin + '/notifications';
-        }
+        const isDev = false;
+        environmentService.baseUrl = window.location.origin;
+        environmentService.notificationsUrl = window.location.origin + '/notifications';
         apiService.setUrls({
             base: isDev ? null : window.location.origin,
             api: isDev ? 'http://localhost:4000' : null,
--- a/src/scss/styles.scss
+++ b/src/scss/styles.scss
@@ -1,5 +1,14 @@
 @import "../css/webfonts.css";
 
+/**** START Bitwarden_RS CHANGES ****/
+/* Hide any link pointing to billing */
+a[href^="#/settings/billing"] {display: none !important; }
+
+/* Hide organization plans */
+app-create-organization > form > div.form-check { display: none !important; }
+app-create-organization > form > h2.mt-5 { display: none !important; }
+/**** END Bitwarden_RS CHANGES ****/
+
 $primary: #3c8dbc;
 $primary-accent: #286090;
 $secondary: #ced4da;
